Dockerfile

```diff
@@ -1,7 +1,24 @@
-FROM node:current
+FROM node:12
 WORKDIR /app
 COPY package.json /app
-RUN npm install
+COPY package-lock.json /app
+RUN npm ci
+RUN mkdir scripts
+# Use "m" to temporarily install mongod in a lightweight way so
+# the asset build task does not encounter problems initializing modules
+# that expect a database. We do not actually need this database in the
+# container, so uninstall it at the end
+RUN npm install -g m
+RUN apt-get -y update
+RUN apt-get -y install scons
+RUN m 3.6
+RUN m tools stable
+# Load balance at least 2 copies
+RUN npm install -g pm2
 COPY . /app
+ENV APOS_MINIFY=1
+ENV NODE_ENV=production
+RUN ./scripts/docker-build-assets-with-temporary-mongod
+RUN m rm 3.6
 EXPOSE 3000
-CMD [ "./scripts/wait-for-it.sh", "mongo:27017", "--", "npm", "start" ]
+CMD ./scripts/docker-start
```

Comment:
We should use array syntax for CMD where possible. The `depends_on` line in docker-compose will tell apostrophe to wait for mongo. We should end up with something like `CMD ["pm2-runtime", "app.js"]` or `CMD ["pm2-runtime", "process.yml"]` based on [PM2 docs](https://pm2.keymetrics.io/docs/usage/docker-pm2-nodejs/).

Reply:
Unfortunately `depends_on` doesn't include any test to verify that the service is truly ready. If that CMD has kicked off, Docker thinks the job is done, but it ain't. This is why `wait-for-it.sh` was born.

"There are limitations to this first solution. For example, it doesn’t verify when a specific service is really ready. "

Reply:
Ahh, gotcha. In that case, I think what we had before was close. We should have `ENTRYPOINT [ "./scripts/wait-for-it.sh", "mongo:27017", "--" ]` and then a `CMD` the same as above. If we need the `CMD` to be a script, that's fine but lets use it in array syntax as well.

Reply:
Also, maybe we should use a environment file instead of hard-coding `mongo:27017` in the entrypoint?

Reply:
mongo:27017 is known from docker-compose, but perhaps docker-compose should inject it into the environment so it's not replicated in Dockerfile.

Reply:
I see what you mean about the utility of ENTRYPOINT. We don't have to use the `--` since we're not passing anything on in the same invocation.

Reply:
> mongo:27017 is known from docker-compose, but perhaps docker-compose should inject it into the environment so it's not replicated in Dockerfile.

That's a good point. Let's do that.